autoload/go/lint.vim

```diff
@@ -55,6 +55,13 @@ function! go#lint#Gometa(autosave, ...) abort
     let cmd += split(g:go_metalinter_command, " ")
   endif
 
+  " gometalinter has a --tests flag to tell its linters whether to run against
+  " tests. While not all of its linters respect this flag, for those that do,
+  " it means if we don't pass --tests, the linter won't run against test
+  " files. One example of a linter that will not run against tests if we do
+  " not specify this flag is errcheck.
+  let cmd += ["--tests"]
```

Comment:
I think it might be better to put this in the `if a:autosave || empty(g:go_metalinter_command)` block above.

That way people who have `g:go_metalinter_command` defined – which is documented as "for users who want to have a complete control over how gometalinter should be executed" – won't have this extra flag, which also addresses your comment and gives people a way to disable it if they really want to.